crates/particles/src/particle.rs

```diff
@@ -0,0 +1,353 @@
+            ParticleKind::EntityEffect => "entity_effect",
+            ParticleKind::ExplosionEmitter => "explosion_emitter",
+            ParticleKind::Explosion => "explosion",
+            ParticleKind::FallingDust(_) => "falling_dust",
+            ParticleKind::Firework => "firework",
+            ParticleKind::Fishing => "fishing",
+            ParticleKind::Flame => "flame",
+            ParticleKind::SoulFireFlame => "soul_fire_flame",
+            ParticleKind::Soul => "soul",
+            ParticleKind::Flash => "flash",
+            ParticleKind::HappyVillager => "happy_villager",
+            ParticleKind::Composter => "composter",
+            ParticleKind::Heart => "heart",
+            ParticleKind::InstantEffect => "instant_effect",
+            ParticleKind::Item(_) => "item",
+            ParticleKind::ItemSlime => "item_slime",
+            ParticleKind::ItemSnowball => "item_snowball",
+            ParticleKind::LargeSmoke => "large_smoke",
+            ParticleKind::Lava => "lava",
+            ParticleKind::Mycelium => "mycelium",
+            ParticleKind::Note => "note",
+            ParticleKind::Poof => "poof",
+            ParticleKind::Portal => "portal",
+            ParticleKind::Rain => "rain",
+            ParticleKind::Smoke => "smoke",
+            ParticleKind::Sneeze => "sneeze",
+            ParticleKind::Spit => "spit",
+            ParticleKind::SquidInk => "squid_ink",
+            ParticleKind::SweepAttack => "sweep_attack",
+            ParticleKind::TotemOfUndying => "totem_of_undying",
+            ParticleKind::Underwater => "underwater",
+            ParticleKind::Splash => "splash",
+            ParticleKind::Witch => "witch",
+            ParticleKind::BubblePop => "bubble_pop",
+            ParticleKind::CurrentDown => "current_down",
+            ParticleKind::BubbleColumnUp => "bubble_column_up",
+            ParticleKind::Nautilus => "nautilus",
+            ParticleKind::Dolphin => "dolphin",
+            ParticleKind::CampfireCosySmoke => "campfire_cosy_smoke",
+            ParticleKind::CampfireSignalSmoke => "campfire_signal_smoke",
+            ParticleKind::DrippingHoney => "dripping_honey",
+            ParticleKind::FallingHoney => "falling_honey",
+            ParticleKind::LandingHoney => "landing_honey",
+            ParticleKind::FallingNectar => "falling_nectar",
+            ParticleKind::Ash => "ash",
+            ParticleKind::CrimsonSpore => "crimson_spore",
+            ParticleKind::WarpedSpore => "warped_spore",
+            ParticleKind::DrippingObsidianTear => "dripping_obsidian_tear",
+            ParticleKind::FallingObsidianTear => "falling_obsidian_tear",
+            ParticleKind::LandingObsidianTear => "landing_obsidian_tear",
+            ParticleKind::ReversePortal => "reverse_portal",
+            ParticleKind::WhiteAsh => "white_ash",
+        }
+    }
+}
+
+unsafe impl Zeroable for ParticleKind {}
+
+unsafe impl Pod for ParticleKind {}
```

Comment:
This is unsound because not all bit patterns are valid for `ParticleKind`. I'd recommend implementing `Serialize + Deserialize` instead; we can transfer particles with `bincode` in `quill`.

Reply:
Ok, will change thanks 👍